### PR TITLE
Check against inconsistencies in URLs of this/previous/latest versions

### PR DESCRIPTION
**Pull request by tripu**
I'm comparing the *contents* (the actual documents) of "previous" vs. "latest"
&hellip;but just the *URLs* of "this" vs. "previous".

I understand that is what will guarantee that we're catching all those tricky situations explained in issue #29&hellip;? I have also extended the tests to try to catch these cases.

(@plehegar, it'd be great if you want to extend the Echidna test suite to cover these cases, once/if this is merged :¬)

### CODE REVIEW

File: lib/rules/headers/dl.js [hunk 1]
```diff
@@ -133,6 +156,21 @@ exports.check = function (sr, done) {
...
+      }
+    }, function() {
+      err('cant-retrieve');
+    });
+
+    if (thisURI.toLowerCase() === previousURI.toLowerCase()) {
+      err('same-this-and-previous');
+    }
+
     done();
```
Comment (astorije) [inline lib/rules/headers/dl.js]: Shouldn't `done()` be inside the `then()` function, since it's asynchronous?

**Comment by astorije** [comment]
On top of my inline comment, I'd like a :+1: from @darobin before giving mine :-)

**Comment by tripu** [comment]
@astorije, you were right, thanks! I think I just fixed this issue, and now `done()` is invoked if/when promises succeed or fail, too.

@darobin, assigning to you as @astorije suggested.

**Comment by astorije** [comment]
@darobin, you have my :+1:, feel free to merge yourself or let me do it!

**Comment by tripu** [comment]
Merged myself after checking with @deniak (and having had @astorije's blessing too)&nbsp;&mdash;&nbsp;this one's been sitting here for too long, and I'd like to have it in master to deploy a new release this week.

**Pull request merged**